PrebidMobile/PrebidMobile-rendering/src/main/java/org/prebid/mobile/rendering/sdk/deviceData/managers/UserConsentManager.java

```diff
@@ -114,6 +108,42 @@ public String getUsPrivacyString() {
         return mUsPrivacyString;
     }
 
+    //fetch advertising identifier based TCF 2.0 Purpose1 value
+    //truth table
+    /*
+                         deviceAccessConsent=true   deviceAccessConsent=false  deviceAccessConsent undefined
+    gdprApplies=false        Yes, read IDFA             No, don’t read IDFA           Yes, read IDFA
+    gdprApplies=true         Yes, read IDFA             No, don’t read IDFA           No, don’t read IDFA
+    gdprApplies=undefined    Yes, read IDFA             No, don’t read IDFA           Yes, read IDFA
+    */
+
+    public boolean canAccessDeviceData() {
+        final String subjectToGdpr = getSubjectToGdpr();
+        final int deviceConsentIndex = 1;
+
+        Boolean gdprApplies = TextUtils.isEmpty(subjectToGdpr) ? null : "1".equals(subjectToGdpr);
```

Comment:
For the future. If you start to use 'final' for some variables it makes sense to apply it everywhere. In this method, the first two variables have 'final' but these two don't have, somehow.